### ISSUE TITLE
Implement Ensemble Information FIlter in ERT

### ISSUE THREAD

**Issue by Blunde1** [equinor/ert#5915]
**Is your feature request related to a problem? Please describe.**
Spurious correlations

**Describe the solution you'd like**
An update algorithm without unnecessary statistical estimation error.

**Describe alternatives you've considered**
Ordinary localization. Tested to be sub-optimal.

**Additional context**
Do not push to ERT at once!
Hijack the Ensemble Smoother to illustrate how an implementation in ERT would work.
Test it on a list of illustrative examples.

**Comment by oyvindeide** [comment]
Yes, will close as part of: https://github.com/equinor/ert/pull/10466
